pbjam/plotting.py

```diff
@@ -284,18 +317,108 @@ def plot_spectrum(self, pg=None, path=None, ID=None, savefig=False):
         ax.set_xlabel(r'Frequency ($\mu \rm Hz$)')
         ax.set_ylabel(r'SNR')
         ax.legend(loc=1)
-
+        
+        fig.tight_layout()
         if savefig:
-            outpath = os.path.join(*[path, f'{type(self).__name__}_{str(ID)}.png'])
-            fig.savefig(outpath)
+            self._save_my_fig(fig, 'spectrum', path, ID)
 
         return fig
 
-    # Asy_peakbag
+
+    def _fill_diag(self, axes, vals, vals_err, idxs):
+        N = int(np.sqrt(len(axes)))
+        axs = np.array(axes).reshape((N,N)).T
+        
+        for i,j in enumerate(idxs):
+            yrng = axs[j,j].get_ylim()
+            
+            v, ve = vals[i], vals_err[i]
+            
+            axs[j,j].fill_betweenx(y=yrng, x1= v-ve[0], x2 = v+ve[-1], color = 'C3', alpha = 0.5)
+    
+    def _plot_offdiag(self, axes, vals, vals_err, idxs):
```

Comment:
Missing docs - same comment as for _fill_diag

Reply:
See comment to _fill_diag